src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/FinancialTransactionCategoryCreateDTO.java

```diff
@@ -1,11 +1,15 @@
 package pl.byczazagroda.trackexpensesappbackend.dto;
 
 import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.regex.RegexConstant;
 
-import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
+import javax.validation.constraints.Pattern;
 
-public record FinancialTransactionCategoryCreateDTO(@NotBlank @Size(max = 30, message = "CATEGORY NAME TOO LONG")
-                                                    String name, @NotNull FinancialTransactionType type,
-                                                    @NotNull Long userId) { }
+public record FinancialTransactionCategoryCreateDTO(
+        @NotNull
+        @Pattern(regexp = RegexConstant.CATEGORY_NAME_PATTERN)
+        String name,
+        @NotNull FinancialTransactionType type,
+        @NotNull Long userId) {
```

Comment:
FTCategoryCreateDTO should not have userId it will be passed separatly to the service. The server will specify the identity of user thats call the api not the user itself

Reply:
You're right, it needs to be changed, but it should be a separate task.